components/meter/meter-circle.js

```diff
@@ -10,11 +10,19 @@ import { RtlMixin } from '../../mixins/rtl/rtl-mixin.js';
  */
 class MeterCircle extends MeterMixin(RtlMixin(LitElement)) {
 	static get styles() {
-		return [ bodyStandardStyles, css`
+		return [ bodySmallStyles, bodyStandardStyles, css`
 		:host {
 			display: inline-block;
 			width: 2.4rem;
 		}
+		:host([text]:not([text-hidden])) {
+			width: fit-content;
```

Comment:
One more thing to highlight re `fit-content`: it wasn't added to FF until v94 ([CanIUse](https://caniuse.com/?search=fit-content)), which is in our "legacy" browser support band. Maybe we're "ok" with the visual not being ideal for those users, but figured I would call it out is case we're not.

Reply:
Abi, what's the visual result of this not working? Just curious how bad it would look.

Reply:
Without the fit-content the text isn't nicely centred under the meter - maybe there is another way to resolve this that I'm unaware of? 
![Screenshot 2024-04-10 at 3 53 33 PM](https://github.com/BrightspaceUI/core/assets/54903661/54995d87-d061-45a8-bab6-f9c79704f94b)

Reply:
Haha what's going on with the other circles in that screenshot? If they're unrelated, then this seems like a fine trade-off for older browsers. As long as it's not totally broken, it's fine.

Reply:
> Haha what's going on with the other circles in that screenshot?

Haha, I think that's just the adjacent meter components included in the screenshot of the demo page.

> Without the fit-content the text isn't nicely centred under the meter - maybe there is another way to resolve this that I'm unaware of?

`d2l-meter-radial` has the same issue. The text simply overflows.

What I observed with `fit-content`: `svg` scales based on the text size. As a result, the same meter would have different size for different languages, or adjacent meters with different text would be different size. That's probably not the behaviour we want.  
![image](https://github.com/BrightspaceUI/core/assets/9042472/dcca4c71-9795-42f7-99da-038a6daf19e2)


I managed to get something sorta working by setting `width: auto` on the host, moving the baked `width: 2.4rem` to the `svg` (from the host), and setting `margin: 0 auto` on the `svg`. But then the text potentially pushes them further apart.
![image](https://github.com/BrightspaceUI/core/assets/9042472/7baa53e1-b566-45f9-a7e2-8611fc00e19c)


But currently, the consumer can set the size by specifying `<d2l-meter-circle style="width: 25%;">`. Moving the `width` to the internal `svg` breaks that behaviour, and so we'd need to do something else to allow them to control the size of the `svg`. Maybe a CSS variable or a property. Nevertheless, it things seem possible, but it's a change in behaviour.

Reply:
used a combo of `classMap` and attribute selectors to get the `width: 2.4rem` appearing on meters without text/ with text-hidden AND don't have a style for width set on the host level :) 

looks like it's working well - I'll add some inline notes for future devs to make it clear what we're doing

![Screenshot 2024-04-11 at 8 49 10 AM](https://github.com/BrightspaceUI/core/assets/54903661/7ac7715b-2d79-4659-9818-f4ec897a7755)

Reply:
https://github.com/BrightspaceUI/core/pull/4587/files#diff-4df82a82ac6364e1408ebd4ce8b7a8ede6a5f0c0e4bb96d0e4f005ebf9f50f33R82-R86

Reply:
It looks like the size of the `svg` would vary based on the size of the text. We can check with Jeff, but my current feeling is that we don't want that behaviour as mentioned above.
![image](https://github.com/BrightspaceUI/core/assets/9042472/0fc29ff7-c778-4f76-abec-2c062fab6289)